src/OrchardCore.Modules/OrchardCore.Facebook/Drivers/FacebookCoreSettingsDisplayDriver.cs

```diff
@@ -57,47 +57,47 @@ public override async Task<IDisplayResult> EditAsync(FacebookCoreSettings settin
                 var protector = _dataProtectionProvider.CreateProtector(FacebookConstants.Features.Core);
 
                 model.AppId = settings.AppId;
-                model.AppSecret = protector.Unprotect(settings.AppSecret);
+                if (!string.IsNullOrWhiteSpace(settings.AppSecret))
+                {
+                    model.AppSecret = protector.Unprotect(settings.AppSecret);
+                }
 
             }).Location("Content:0").OnGroup(FacebookConstants.Features.Core);
         }
 
         public override async Task<IDisplayResult> UpdateAsync(FacebookCoreSettings settings, BuildEditorContext context)
         {
-            if (context.GroupId == FacebookConstants.Features.Core)
```

Comment:
If we remove this check, the UpdateAsync executes when saving the FaceBookLoginSettings and sets validation errors in the login settings view